Core/MIPS/ARM64/Arm64IRCompVec.cpp

```diff
@@ -84,19 +84,322 @@ void Arm64JitBackend::CompIR_VecArith(IRInst inst) {
 	}
 }
 
+enum class Arm64Shuffle {
+	DUP0_AAAA,
+	DUP1_BBBB,
+	DUP2_CCCC,
+	DUP3_DDDD,
+	MOV_ABCD,
+	TRN1_AACC,
+	TRN2_BBDD,
+	UZP1_ACAC,
+	UZP2_BDBD,
+	ZIP1_AABB,
+	ZIP2_CCDD,
+	REV64_BADC,
+	EXT4_BCDA,
+	EXT8_CDAB,
+	EXT12_DABC,
+
+	// These steps are more expensive and use a temp.
+	REV64_EXT8_CDBA,
+	REV64_EXT8_DCAB,
+	EXT4_UZP1_BDAC,
+	EXT4_UZP2_CABD,
+	EXT8_ZIP1_ACBD,
+	EXT8_ZIP2_CADB,
+
+	// Any that mutate dest must be after this point.
```

Comment:
Mutate in the sense of "partially change" instead of replace, I assume? All of these mutate dest in some way :)

Reply:
I can reword, but I meant as opposed to fully setting it.  XTN2, INS, etc. require that a MOV happened first or etc.

-[Unknown]

Reply:
Yeah. Eh, don't bother changing it, it's kinda evident from context which meaning of mutate it is..